libmariadb/mariadb_lib.c

```diff
@@ -849,83 +886,114 @@ static size_t rset_field_offsets[]= {
   OFFSET(MYSQL_FIELD, org_name_length)
 };
 
+
+/**
+  Read field metadata from field descriptor and store it in MYSQL_FIELD structure.
+  String values in MYSQL_FIELD are allocated in a given allocator root.
+
+  @param mysql          connection handle
+  @param alloc          memory allocator root
+  @param default_value  flag telling if default values should be read from
+                        descriptor
+  @param server_capabilities  protocol capability flags which determine format of
+                              the descriptor
+  @param row            field descriptor
+  @param field          address of MYSQL_FIELD structure to store metadata in.
+
+  @returns 0 on success.
+*/
+
+int
+unpack_field(const MYSQL *mysql, MA_MEM_ROOT *alloc, my_bool default_value,
```

Comment:
This is extracted as is from `unpack_fields` function below so it can be called directly when reading packet metadata similar to mysql source code [here](https://github.com/mysql/mysql-server/blob/bd9c260cf7026a54d9bf91ce6782cdb4e6a25c71/sql-common/client.c#L2036)